eclipse-scout-core/src/tree/Tree.ts

```diff
@@ -2501,6 +2501,14 @@ export class Tree extends Widget implements TreeModel, Filterable<TreeNode> {
       this._removeNodes(deletedNodes, parentNode || parentNodesToReindex);
     }
 
+    arrays.ensure(parentNode || parentNodesToReindex).forEach(p => {
+      if (p?.expandedLazy && !p.childNodes?.length) {
```

Comment:
Why are `?` necessary? `parentNodesToReindex` is always an array and only contains valid nodes. `parentNode` might be null or undefined, but this is removed automatically by arrays.ensure(). `TreeNode#childNodes` is also never null.